tests/lava/proc/sparse/test_process.py

```diff
@@ -0,0 +1,99 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+import unittest
+import numpy as np
+from scipy.sparse import csr_matrix, spmatrix
+
+from lava.utils.sparse import find_with_explicit_zeros
+from lava.proc.sparse.process import Sparse, LearningSparse, DelaySparse
+from lava.proc.learning_rules.stdp_learning_rule import STDPLoihi
+
+
+class TestFunctions(unittest.TestCase):
+    """Test helper function for Sparse"""
+
+    def test_find_with_explicit_zeros(self):
+
+        mat = np.random.randint(-10, 10, (3, 5))
+        spmat = csr_matrix(mat)
+        spmat.data[0] = 0
+
+        dst, src, vals = find_with_explicit_zeros(spmat)
+
+        self.assertTrue(np.all(spmat.data in vals))
+
+
+class TestSparseProcess(unittest.TestCase):
+    """Tests for Sparse class"""
+
+    def test_init(self):
+        """Tests instantiation of Sparse"""
+        shape = (100, 200)
+        weights = np.random.random(shape)
+
+        # Sparsify
+        weights[weights < 0.7] = 0
+
+        # Convert to spmatrix
+        weights_sparse = csr_matrix(weights)
+
+        conn = Sparse(weights=weights_sparse)
+
+        self.assertIsInstance(conn.weights.init, spmatrix)
+        np.testing.assert_array_equal(conn.weights.init.toarray(), weights)
+
+
+class TestLearningSparseProcess(unittest.TestCase):
```

Comment:
What about negative tests? You are not testing the validation method, for instance.

Reply:
Removed the validation method as it's useless. A spmatrix can only be 2d which is exactly the thing we validated.